lib/seahorn/BvOpSem2.cc

```diff
@@ -1503,7 +1511,7 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemBase {
       setValue(inst, havoc(inst));
 
       m_ctx.setMemReadRegister(memIn);
-      m_ctx.setMemWriteRegister(memOut);
+      m_ctx.setMemWriteRegister(memIn);
```

Comment:
the old code was correct. The current one  will generate a constraint `MEM_IN = MEM_IN with a change`. The equality is trivially false, which will cause everything else to be unsat